src/exts/tvdb_info/ui.py

```diff
@@ -0,0 +1,485 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, TypedDict, override
+
+import discord
+
+from src.bot import Bot
+from src.db_adapters import list_put_item, list_remove_item, refresh_list_items, user_get_list_safe, user_get_safe
+from src.db_tables.user_list import UserList, UserListItem, UserListItemKind
+from src.settings import THETVDB_COPYRIGHT_FOOTER, THETVDB_LOGO
+from src.tvdb import Movie, Series
+from src.utils.iterators import get_first
+from src.utils.log import get_logger
+from src.utils.tvdb import by_season
+
+if TYPE_CHECKING:
+    from src.db_tables.user import User
+    from src.tvdb.client import Episode
+
+log = get_logger(__name__)
+
+MOVIE_EMOJI = "🎬"
+SERIES_EMOJI = "📺"
+
+
+class ReactiveButtonState(TypedDict):
+    """Type definition for a reactive button state."""
+
+    label: str
+    style: discord.ButtonStyle
+    emoji: str
+
+
+# A reactive button is a tuple of the button, the item in the list, the state when active, and the state when inactive.
+# It allows us to programmatically change the button's appearance based on the state of the item in the list.
+type ReactiveButton = tuple[
+    discord.ui.Button,  # pyright: ignore[reportMissingTypeArgument]
+    UserListItem | None,
+    ReactiveButtonState,
+    ReactiveButtonState,
+]
```

Comment:
I'd prefer a `typing.NamedTuple` for this then a simple tuple type alias. That way it will be cleaner which of the fields are for what, especially for the last 2 items there (active_state, inactive_state), it's very easy to use the other one by mistake when you're just indexing.